src/eos-google-chrome-installer

```diff
@@ -0,0 +1,241 @@
+
+    def _set_as_default_browser(self):
+        mimetypes = ["text/html",
+                     "x-scheme-handler/http",
+                     "x-scheme-handler/https",
+                     "x-scheme-handler/about"]
+        for mimetype in mimetypes:
+            try:
+                subprocess.Popen(["xdg-mime", "default", "google-chrome.desktop", mimetype])
+            except OSError:
+                exit_with_error("Couldn't start xdg-mime. Are xdg-utils installed?")
+        try:
+            subprocess.Popen(["xdg-settings", "set", "default-web-browser", "google-chrome.desktop"])
+        except OSError:
+            exit_with_error("Couldn't start xdg-settings. Are xdg-utils installed?")
+
+    def _touch_done_file(self):
+        chrome_setup_done_path = pathlib.Path(os.path.expanduser("~/.config/google-chrome-initial-setup-done"))
+        if not chrome_setup_done_path.exists():
+            chrome_setup_done_path.touch()
+
+    def _disable_gsetting(self):
+        schema_source = Gio.SettingsSchemaSource.get_default()
+        if not schema_source.lookup(self.GSETTINGS_SCHEMA, True):
+            return False
+
+        schema = Gio.Settings(self.GSETTINGS_SCHEMA)
+        schema.set_boolean("enabled", False)
+
+    def _post_install_chrome(self):
+        self._set_as_default_browser()
+        self._touch_done_file()
+        self._disable_gsetting()
+
+    def _run_app_center_for_chrome(self):
+        # We use the APP ID as the one for GNOME Software, to let it choose the best one.
+        chrome_app_center_id = self.FLATPAK_CHROME_APP_ID
+
+        # FIXME: Ideally, we should be able to pass 'com.google.Chrome' to GNOME Software
+        # and it would do the right thing by opening the page for the app's branch matching
+        # the default branch for the apps' source remote. Unfortunately, this is not the case
+        # at the moment and fixing it is non-trivial, so we'll construct the full unique ID
+        # that GNOME Software expects, right from here, based on the remote's metadata.
+        default_branch = None
+        try:
+            remote = self._installation.get_remote_by_name(self.FLATPAK_REMOTE_EOS_APPS)
+        except GLib.Error as e:
+            logging.warning("Could not determine default branch for remote %s", self.FLATPAK_REMOTE_EOS_APPS)
+
+        default_branch = remote.get_default_branch()
+        if default_branch:
+            chrome_app_center_id = 'system/flatpak/{}/desktop/{}.desktop/{}'.format(self.FLATPAK_REMOTE_EOS_APPS,
+                                                                                    self.FLATPAK_CHROME_APP_ID,
+                                                                                    default_branch)
+        if self._initial_setup:
+            app_center_argv = ['gnome-software', '--install', chrome_app_center_id, '--interaction', 'none']
+        else:
+            app_center_argv = ['gnome-software', '--details={}'.format(chrome_app_center_id)]
```

Comment:
Sorry I have not noticed this before but you need an early return here, not to hook into the wait + post-install logic in case we are not in the "initial setup" mode.

I would add an extra empty line right before the `try` below for clarity, too.